x-pack/plugins/observability_solution/infra/public/components/ml/anomaly_detection/anomalies_table/anomalies_table.tsx

```diff
@@ -134,13 +152,55 @@ const AnomalyActionMenu = ({
   ];
 
   if (!disableShowInInventory) {
-    items.push(
-      <EuiContextMenuItem key="showInInventory" icon="search" onClick={showInInventory}>
+    const buildFilter = buildCombinedAssetFilter({
+      field: HOST_NAME_FIELD,
+      values: influencers,
+      dataView: metricsView?.dataViewReference,
+    });
+
+    let newFilter: Filter[] = [];
+    if (!Array.isArray(buildFilter)) {
+      newFilter = [buildFilter];
+    }
+
+    const showInHostsItem = !hostName ? (
+      <EuiContextMenuItem
+        key="showAffectedHosts"
```

Comment:
It is interesting that `EuiContextMenuPanel` doesn't add a key to items.

Reply:
Honestly, I followed the implementation of existing 'Show in Inventory' button. 

Indeed, `EuiContextMenuPanel` generates unique key. Just confirmed that. Updated. Thx!
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/2dd358e7-c054-4cf9-a4b9-ee7b28c08cc4">